cli/test/e2e/package_authoring_e2e_test.go

```diff
@@ -163,89 +193,413 @@ spec:
+
+			expectedPackageBuild := strings.TrimSpace(replaceSpaces(testcase.ExpectedPkgBuild))
+			out = clearKeys(keysToBeIgnored, strings.TrimSpace(replaceSpaces(out)))
+			require.Equal(t, expectedPackageBuild, out, "Expected PackageBuild output to match")
 
-		kappCtrl.RunWithOpts([]string{"pkg", "release", "--version", "1.0.0", "--tty=true", "--chdir", workingDir},
-			RunOpts{NoNamespace: true, StdinReader: promptOutput.StringReader(),
-				StdoutWriter: promptOutput.BufferedOutputWriter(), Interactive: true})
+			// Verify package resources
+			out, err = readFile("package-resources.yml")
+			require.NoErrorf(t, err, "Expected to read package-resources.yml")
 
-		keysToBeIgnored := []string{"creationTimestamp:", "releasedAt:", "image"}
+			expectedPackageResources := strings.TrimSpace(replaceSpaces(testcase.ExpectedPkgResource))
+			out = clearKeys(keysToBeIgnored, strings.TrimSpace(replaceSpaces(out)))
+			require.Equal(t, expectedPackageResources, out, "Expected package resources output to match")
 
-		// Verify PackageMetadata artifact
-		out, err := readFile("./carvel-artifacts/packages/testpackage.corp.dev/metadata.yml")
-		require.NoErrorf(t, err, "Expected to read metadata.yml")
+			// Verify vendir
+			out, err = readFile("vendir.yml")
+			require.NoErrorf(t, err, "Expected to read vendir.yml")
 
-		expectedPackageMetadata = strings.TrimSpace(replaceSpaces(expectedPackageMetadata))
-		out = clearKeys(keysToBeIgnored, strings.TrimSpace(replaceSpaces(out)))
-		require.Equal(t, expectedPackageMetadata, out, "Expected PackageMetadata to match")
+			expectedVendirOutput := strings.TrimSpace(replaceSpaces(testcase.ExpectedVendir))
+			out = clearKeys(keysToBeIgnored, strings.TrimSpace(replaceSpaces(out)))
+			require.Equal(t, expectedVendirOutput, out, "Expected vendir output to match")
+		})
 
-		// Verify Package artifact
-		out, err = readFile("./carvel-artifacts/packages/testpackage.corp.dev/package.yml")
-		require.NoErrorf(t, err, "Expected to read package.yml")
+		logger.Section(fmt.Sprintf("%s: Package release", testcase.Name), func() {
+			releaseInteraction := Interaction{
+				Prompts: []string{"Enter the registry URL"},
+				Inputs:  []string{env.Image},
+			}
 
-		expectedPackage = strings.TrimSpace(replaceSpaces(expectedPackage))
-		out = clearKeys(keysToBeIgnored, strings.TrimSpace(replaceSpaces(out)))
-		require.Equal(t, expectedPackage, out, "Expected Package to match")
-	})
+			go releaseInteraction.Run(promptOutput)
+
+			kappCtrl.RunWithOpts([]string{"pkg", "release", "--version", "1.0.0", "--tty=true", "--chdir", workingDir},
+				RunOpts{NoNamespace: true, StdinReader: promptOutput.StringReader(),
+					StdoutWriter: promptOutput.BufferedOutputWriter(), Interactive: true})
+
+			keysToBeIgnored := []string{"creationTimestamp:", "releasedAt:", "image"}
+
+			// Verify PackageMetadata artifact
+			out, err := readFile("./carvel-artifacts/packages/testpackage.corp.dev/metadata.yml")
```

Comment:
nit: could we use variables/constants for stuff we are using repetitively?
Maybe variables like `packageDir` and `pkgName`